src/node_config.rs

```diff
@@ -80,39 +116,61 @@ impl BootstrapNode {
 }
 
 /// Config parsed from command line arguments.
-#[derive(Clone, PartialEq, Eq, Debug)]
-pub struct CliConfig {
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize)]
+pub struct NodeConfig {
     /// UDP address to run DHT node
+    #[serde(rename = "udp-address")]
+    #[serde(default)]
     pub udp_addr: Option<SocketAddr>,
     /// TCP addresses to run TCP relay
+    #[serde(rename = "tcp-addresses")]
+    #[serde(default)]
     pub tcp_addrs: Vec<SocketAddr>,
     /// DHT SecretKey
+    #[serde(rename = "secret-key")]
```

Comment:
What if both `secret-key` and `keys_file` are specified?

Reply:
Nice catch! It should be `#[serde(skip_deserializing)]`. If a client uses `config` then it should use `keys-file`, not `secret-key`

Reply:
done